test/fakes/fakemetrics/fakemetrics.go

```diff
@@ -56,7 +56,12 @@ func (m *FakeMetrics) SetGauge(key []string, val float32) {
 func (m *FakeMetrics) SetGaugeWithLabels(key []string, val float32, labels []telemetry.Label) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	m.metrics = append(m.metrics, MetricItem{Type: SetGaugeWithLabelsType, Key: key, Val: val, Labels: labels})
+	m.metrics = append(m.metrics, MetricItem{
+		Type:   SetGaugeWithLabelsType,
+		Key:    key,
+		Val:    val,
+		Labels: telemetry.GetSanitizedLabels(labels),
```

Comment:
One last (small) comment: this (and the other call sites) would be a little easier to read and more idiomatic if this function was named `telemetry.SanitizeLabels` instead

Everything else here looks great